### PR TITLE
Chore/cm model layouts

### PR DESCRIPTION
**Pull request by alexandrebodin**
#### Description of what you did:

Allow custom default model layouts and metadatas with files.

Fix plugin model overwrites in extensions

#### My PR is a:

- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [ ] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [ ] Documentation
- [ ] Framework
- [ ] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

File: packages/strapi-plugin-content-manager/services/utils/configuration/index.js [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
+} = require('../../../controllers/validation');
+
+async function validateCustomConfig(model, schema) {
+  try {
+    await createModelConfigurationSchema(model, schema, {
+      allowUndefined: true,
+    }).validate(model.config);
+  } catch (error) {
+    throw new Error(
+      `Invalid Model configuration for model ${model.uid} Verify your Model.config.js(on) file:\n  - ${error.message}\n`
```
Comment (derrickmehaffy) [inline packages/strapi-plugin-content-manager/services/utils/configuration/index.js]: mmmm what is this? `Model.config.js`

File: packages/strapi-plugin-users-permissions/models/User.config.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  metadatas: {
+    resetPasswordToken: {
```
Comment (derrickmehaffy) [inline packages/strapi-plugin-users-permissions/models/User.config.js]: should probably also add provider here

Comment (alexandrebodin) [reply packages/strapi-plugin-users-permissions/models/User.config.js]: Provider is currently there so I'll leave it as is for now :)

File: packages/strapi-plugin-content-manager/services/utils/configuration/index.js (hunk 1 above)
Comment (alexandrebodin) [reply packages/strapi-plugin-content-manager/services/utils/configuration/index.js]: Something new (e.g new User.config.js)

File: packages/strapi-plugin-users-permissions/models/User.config.js (hunk 2 above)
Comment (derrickmehaffy) [reply packages/strapi-plugin-users-permissions/models/User.config.js]: I mean you shouldn't be able to edit it (or see it on the edit view)

Comment (alexandrebodin) [reply packages/strapi-plugin-users-permissions/models/User.config.js]: Yeah I guess you are right this just isn't used anyway in the controller. I'll remove it too then

**Pull request merged**